x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/esql/build_esql_search_request.ts

```diff
@@ -61,5 +61,6 @@ export const buildEsqlSearchRequest = ({
         filter: requestFilter,
       },
     },
+    wait_for_completion_timeout: '4m', // hard limit request timeout is 5m set by ES proxy and alerting framework. So, we should be fine to wait 4m for async query completion. If rule execution is shorter than 4m and query was not completed, it will be aborted.
```

Comment:
setting `wait_for_completion_timeout` this high makes this effectively a synchronous query. Should this be [`keep_alive`](https://www.elastic.co/guide/en/elasticsearch/reference/current/esql-async-query-api.html#esql-async-query-api-request-body) instead? Some keep_alive value longer than the rule timeout would be sufficient. I think we want `wait_for_completion_timeout` to be some number of seconds, like 5 or 10 seconds.

Setting `keep_alive` will help ensure that the results are deleted quickly even if the cleanup `DELETE` request fails.

Reply:
I see, that makes sense. Is 4m the right value for both ECH and serverless? The default rule timeout is different for serverless, right? Is the default connection timeout different?

I think it would be worth following up on this PR to see if we can get access to the rule timeout or maybe have the framework wrapper inject a `keep_alive` value depending on the rule timeout. There could be some use case eventually for async requests initiated by one rule execution and retrieved by a later one, so maybe not every async request needs to have keep_alive == rule timeout, but I think it's the typical scenario for us. It'll just help with system resilience to set it since there could be hundreds of ESQL rules running every few minutes and we'll never need those results for 5 days.

Reply:
> I see, that makes sense. Is 4m the right value for both ECH and serverless? The default rule timeout is different for serverless, right? Is the default connection timeout different?


Default rule timeout for serverless is 1m. If async query request runs longer that this, it would be aborted and query cancelled/deleted in ES.
It's 5m for ECH. If query takes longer than this to finish, it would be cancelled and deleted as well

Reply:
@marshallmain , issue to expose rule timeout to rule executor: https://github.com/elastic/kibana/issues/218072